### PR TITLE
Adding user agent marker for related records

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

**Comment by huminzhi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 80%</td>
    </tr>
  </tbody>
</table>

## Current coverage for SalesforceSDKCore.framework is `48%`
Files changed | - | - 
--- | --- | ---
SFPushNotificationManager.m | `28%` | :no_entry_sign:

---
## Current coverage for SalesforceAnalytics.framework is `41%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SmartSync/SmartSync/Classes/Util/SFParentChildrenSyncHelper.m [hunk 1]
```diff
@@ -64,12 +66,11 @@ + (NSString*) getNonDirtyRecordIdsSql:(SFParentInfo*)parentInfo childrenInfo:(SF
 }
 
 + (void)saveRecordTreesToLocalStore:(SFSmartSyncSyncManager *)syncManager target:(SFSyncTarget *)target parentInfo:(SFParentInfo *)parentInfo childrenInfo:(SFChildrenInfo *)childrenInfo recordTrees:(NSArray *)recordTrees {
-
+    [SFSDKAppFeatureMarkers registerAppFeature:kSFAppFeatureRelatedRecords];
```
Comment (wmathurin) [inline libs/SmartSync/SmartSync/Classes/Util/SFParentChildrenSyncHelper.m]: If we don't care about how often they sync up or sync down related records, shouldn't we simply put the registerAppFeature call in the constructor for parent-sync-down-target and parent-sync-up-target?

**Comment by bhariharan** [comment]
@wmathurin Done [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2056/commits/0d895c65bc842f956ac0466837ca0d6d6c5f4143). Could you add the same marker to the sync up constructor when you tackle that story?

**Comment by wmathurin** [comment]
Sure, will do.

**Pull request merged**